src/schema.js

```diff
@@ -202,16 +214,31 @@ export default class DataStructureGenerator {
         element.description = new StringElement(schema.description);
       }
 
-      if (schema.default !== undefined && !_.isArray(schema.default) &&
-          !_.isObject(schema.default)) {
+      let def = schema.default;
+
+      if (def !== undefined && !_.isArray(def) && !_.isObject(def)) {
         // TODO Support defaults for arrays and objects
-        element.attributes.set('default', schema.default);
+        if (schema.enum) {
+          def = this.wrapEnum(this.minim.toElement(def));
+        }
+
+        element.attributes.set('default', def);
       }
 
+      let samples = [];
+
       if (schema.examples) {
-        element.attributes.set('samples', schema.examples);
+        samples = schema.examples;
       } else if (schema.example) {
-        element.attributes.set('samples', new ArrayElement([schema.example]));
+        samples = new ArrayElement([schema.example]);
```

Comment:
In other cases above the `samples` will be a JavaScript array instead of an `ArrayElement`. Can we make this consistent? The `ArrayElement` will get converted back to a normal array due to the `map` below too.

Little bit more confusing when `samples` can be either an array or `ArrayElement` and makes it easier to introduce bugs at a later date when it could be either types.